api/app/controllers/spree/api/v1/checkouts_controller.rb

```diff
@@ -25,6 +25,11 @@ def advance
         end
 
         def update
+          if @order.completed?
```

Comment:
According to the docs this method should return the order even if no changes were made. If the order is already completed then the "authorize! :update @order, order_token" line will cause exception, i added the completed? check to prevent that.